topology/probes/istio/istio.go

```diff
@@ -49,15 +52,22 @@ func NewIstioProbe(g *graph.Graph) (*k8s.Probe, error) {
 		return nil, err
 	}
 
-	subprobes := map[string]k8s.Subprobe{
-		"destinationrule":  newDestinationRuleProbe(client, g),
-		"gateway":          newGatewayProbe(client, g),
-		"quotaspec":        newQuotaSpecProbe(client, g),
-		"quotaspecbinding": newQuotaSpecBindingProbe(client, g),
-		"serviceentry":     newServiceEntryProbe(client, g),
-		"virtualservice":   newVirtualServiceProbe(client, g),
+	resourceHandlers := map[string]resourceHandler{
+		"destinationrule":  newDestinationRuleProbe,
+		"gateway":          newGatewayProbe,
+		"quotaspec":        newQuotaSpecProbe,
+		"quotaspecbinding": newQuotaSpecBindingProbe,
+		"serviceentry":     newServiceEntryProbe,
+		"virtualservice":   newVirtualServiceProbe,
+	}
+
+	subprobes := make(map[string]k8s.Subprobe)
+	for name, handler := range resourceHandlers {
+		subprobes[name] = handler(client, g)
```

Comment:
All the subprobes are created even if they are not enabled

Reply:
ok - have corrected